text/0000-fabric-private-chaincode-lite.md

```diff
@@ -240,129 +238,138 @@ The FPC-Lite architecture is constituted by a set of components which are design
-
+The operation is performed by executing the `initEnclave` admin command (see step 2. above).
 The command can be triggered through the FPC Client SDK (see [earlier Deployment Section](#deployment)).
 <!--
 Alternatively, the administrator can use the Fabric client to:
 query the `initEnclave` FPC chaincode function;
 pass the result as an argument in the invocation to the `registerEnclave` Enclave Registry chaincode function.
 -->
-A successful initialization and registration will result in a new entry (the enclave credentials, which include the chaincode definition, attestation and key material) in the registry namespace on the ledger.
 
-Internally, the command operates as follows.
-First, it issues an `initEnclave` query which reaches the FPC Shim.
-The shim initializes the enclave (which will process FPC chaincode invocations) with the chaincode parameters received from the peer, namely: the chaincode definition and the channel identifier.
-Then, it generate the key material: public/private key-pairs for signing and encryption, and the state encryption key.
-The public signing key is used as enclave identifier.
-The shim completes the execution by returning the `credentials` of the FPC Chaincode: all public parameters and a hardware-based remote attestation covering them.
-Importantly, the code identity (or hash of the FPC Chaincode) specified in the attestation should match the version in the chaincode definition.
+A successful initialization and registration will result in a new entry in the enclave registry namespace on the ledger. In particular, each entry contains enclave credentials, which cryptographically bind the enclave to the chaincode as defined in the chaincode definition.
 
-In the case of EPID-based Intel® SGX, attestations must be converted into a publicly-verifiable evidence by contacting IAS.
-The FPC Client SDK performs this step using the admin's IAS subscription.
+Internally, the initialization command operates as follows:
 
+First, it issues an `initEnclave` query which reaches the FPC Shim.
+The shim initializes (creates) the enclave (which will process FPC chaincode invocations) with the chaincode parameters received from the peer, namely: the chaincode definition and the channel identifier.
+Then, the enclave generates public/private key-pairs for signing and encryption. The public signature key is used as a enclave identifier.  The key material for chaincode state encryption is generated in a subsequent key generation protocol and we refer to [FPC specification](../images/fpc/full-detail/fpc-key-dist.png) for more details.
+The enclave initialization completes by returning the `credentials` of the FPC Chaincode.
+The credentials contain all public chaincode parameters, including the enclave public signature key.
+In particular, these information are protected through the process of an attestation protocol.
+This protocol may be different depending on the TEE technology used. 
```

Comment:
I would drop this. From our attestation API design we actually did consider some other protocols  (e.g., DCAP) and they map to the flows. If something largely deviates it very likely will also require changes to other flows involving TEE creation and invocation, not only attestation specifc and that seems to go way beyond FPC Lite?
```suggestion
```